### ISSUE TITLE
"Convert identifier to uppercase" code assist doesn't replace references to the variable

### ISSUE THREAD

**Issue by MariaSolOs** [rust-lang/rust-analyzer#15319]
**rust-analyzer version**: 0.3.1591-standalone (d82451103 2023-07-16)

**rustc version**: rustc 1.72.0-nightly (37998ab50 2023-06-11)

Given the following code:
```rust
    const should_foo: bool = true;
    if should_foo {
        println!("foo");
    } else {
        println!("bar");
    }
```
`rust-analyzer` will display the "Convert the identifier to upper case" code assist. However, when applying the refactoring only the declaration is updated, not the references.

And so the outcome is:

```rust
    const SHOULD_FOO: bool = true;
    if should_foo { // error
        println!("foo");
    } else {
        println!("bar");
    }
```

**Comment by lnicola** [comment]
Are you sure about this? Our diagnostic doesn't trigger for inner items, but otherwise the fix works just fine:

![image](https://github.com/rust-lang/rust-analyzer/assets/308347/19dd06c5-2fa9-4179-afe8-1f2975d13f75)

![image](https://github.com/rust-lang/rust-analyzer/assets/308347/9b1dddcb-79a7-4fc2-81c4-ce5720764060)

Then there's also the warning from `rustc` which doesn't work properly, CC #7742:

![image](https://github.com/rust-lang/rust-analyzer/assets/308347/d34610de-dd5f-4aa6-9cab-bde9e730c93b)

**Comment by lowr** [comment]
The diagnostic mentioned in OP does come from rustc, which doesn't account for name references, but we should be also reporting the diagnostics for them anyway. #15320 should fix that.

**Comment by MariaSolOs** [comment]
And as always, I’m impressed by the bug fixing speed of y’all. Thanks a lot!

**Comment by MariaSolOs** [comment]
@lowr @lnicola Hello again! Quick question: With the latest version of RA I'm now seeing the 2 code actions below:

<img width="760" alt="image" src="https://github.com/rust-lang/rust-analyzer/assets/62502207/18d7da56-6c83-45b5-9def-8308b38e6475">

The first one does correctly update references, while the 2nd one (which is the one I originally mentioned in this issue) still only updates the declaration. I'm not asking for further changes, I just want to better understand the difference between them.

**Comment by MariaSolOs** [comment]
In particular, from the description of #15320 it sounds like RA just started providing those refactorings, and so who's providing the 2nd one? Because from the UI above it looks like RA is the only code action provider...

**Comment by lnicola** [comment]
It's from rustc (cargo check), see my comment above.

**Comment by MariaSolOs** [comment]
> It's from rustc (cargo check), see my comment above.

Oh I'm so sorry, I missed that very important last sentence. My bad!
